### PR TITLE
Remove non-admin feed generation

### PR DESCRIPTION
**Pull request by cmpilato**
Administrators should have full control of when their RSS feeds are
generated or refreshed, especially as this operation can be somewhat
filesystem-intensive (aggregating potentially thousands of XML
metadata files).  The feed.xml file is already updated after
administrative actions, can be updated via a private cron URL, and can
be updated explicitly and on-demand through the administrative
console.  That is sufficient.  There's simply no reason to allow every
anonymous page view to perform this function, not to allow other
read-only feed requests to do so.

In other words, PodcastGenerator *should* be able to run in static
mode on a read-only filesystem.

### CODE REVIEW

**Comment by emilengler** [comment]
The purpose of this is to list future episodes once their pubdate is reached.

**Comment by cmpilato** [comment]
> The purpose of this is to list future episodes once their pubdate is reached.

Hrm...  I would argue that that is _precisely_ the value that the cron trigger brings to this picture.

Essentially, the current approach of this codebase asks every individual podcast site visitor (and indeed, nearly every click of those visitors) to pay a performance penalty merely to automate a use-case that is, by comparison, a vastly more rare event.

To put numbers to it, imagine that every day for the past two years, a popular podcast posts a few episodes that are post-dated for the next day.  In order to save that administrator from running a cron job **once per day** so new episodes can go live, you are asking every one of his daily site visitors (100?  1,000?  10,000?) to wait while 700+ XML files are parsed and amalgamated into a new `feed.xml` file that is exactly identical to the previous feed.xml file 99.99% of the time.

This is a wildly lopsided tradeoff which is trivially avoidable thanks to the rather clever private cron feature.

Anyway, I've updated my pull request to restore the generateRSS() call for feed.php.  That should bring things back into line with the PG2.7 codebase, which did not generate the feed on every index.php page visit, but did allow the cron and feed endpoints to do so.

**Comment by emilengler** [comment]
> Hrm... I would argue that that is precisely the value that the cron trigger brings to this picture.

Probably most users don't know what a crond is nor they have a server to run it. I guess most PG users use webspaces.

However I agree on that a new feed on every site call is useless

**Comment by cmpilato** [comment]
Wondering how this was overlooked for v3.0.1?

**Comment by emilengler** [comment]
I see this more as a feature rather than a bug and 3.0.1 should be a bugfix release only
